dapr_agents/agents/durable.py

```diff
@@ -59,71 +71,162 @@ def __init__(
-            execution_config=execution_config,
+            state=state,
+            memory=memory,
+            registry=registry,
+            execution=execution,
             agent_metadata=agent_metadata,
+            workflow_grpc=workflow_grpc,
             llm=llm,
             tools=tools,
             prompt_template=prompt_template,
         )
 
+        apply_grpc_options(self.workflow_grpc_options)
+
         self._runtime: wf.WorkflowRuntime = runtime or wf.WorkflowRuntime()
         self._runtime_owned = runtime is None
         self._registered = False
         self._started = False
 
+        self.output_type = None
+        if profile and getattr(profile, "output_type", None) is not None:
+            self.output_type = profile.output_type
+
+        # Maps handoff tool aliases to their downstream targets and spec metadata
+        self._handoff_tools: Dict[str, str] = {}
+        self._handoff_catalog: Dict[str, HandoffSpec] = {}
+
+        # Ensure tools container is always a list for efficient append operations
+        if not isinstance(self.tools, list):
+            self.tools = list(self.tools or [])
+
+        normalized_specs: list[HandoffSpec] = []
+        if handoffs:
+            for spec in handoffs:
+                if not isinstance(spec, HandoffSpec):
+                    raise TypeError(
+                        "handoffs iterable must contain HandoffSpec instances."
+                    )
+                normalized_specs.append(spec)
+
+        if normalized_specs:
+
+            def metadata_resolver() -> Dict[str, Any]:
+                return self.get_agents_metadata(
+                    exclude_self=True,
+                    exclude_orchestrator=True,
+                )
+
+            for spec in normalized_specs:
+                self._register_handoff_tool(spec, metadata_resolver)
+
+    # ------------------------------------------------------------------
+    # Handoff tool registration
+    # ------------------------------------------------------------------
+
+    def _register_handoff_tool(
+        self,
+        spec: HandoffSpec,
+        metadata_resolver: Callable[[], Mapping[str, Any]],
+    ) -> None:
+        """
+        Materialize a handoff tool, register it with the executor, and track aliases.
+        """
+        tool = create_handoff_tool(
+            target_agent_name=spec.agent_name,
+            source_agent_name=self.name,
+            metadata_resolver=metadata_resolver,
+            tool_name=spec.tool_name,
+            description=spec.description,
+            input_model=spec.input_model,
+        )
+
+        try:
+            self.tool_executor.register_tool(tool)
+        except AgentToolExecutorError:
+            logger.warning("Duplicate handoff tool skipped: %s", tool.name)
+            return
+
+        self.tools.append(tool)
+
+        alias_keys = {
+            alias.lower()
+            for alias in generate_handoff_aliases(tool.name, spec.agent_name)
```

Comment:
why do we need a handoff alias? if we register an agent/tool name, then don't we have all the info we need to not need an alias? am i missing something here?